### PR TITLE
during eval revert to vocab when no cands in ranker

### PR DESCRIPTION
**Pull request by alexholdenmiller**
error flagged in #1271, during interactive mode no candidates are available so model crashes

this way model can at least send one-token replies during eval to attempt to answer the question

during training, if a batch happens to have no labels, skip it (some datasets have a final text-only example at the end of episodes with no label)

### CODE REVIEW

File: parlai/core/torch_agent.py [hunk 1]
```diff
@@ -545,7 +545,8 @@ def _set_label_cands_vec(self, obs, add_start, add_end, truncate):
                 vecs = obs['label_candidates_vecs']
                 for i, c in enumerate(vecs):
                     vecs[i] = self._check_truncate(c, truncate)
-        elif self.rank_candidates and 'label_candidates' in obs:
+        elif self.rank_candidates and obs.get('label_candidates'):
+            # import pdb; pdb.set_trace()
```
Comment (stephenroller) [inline parlai/core/torch_agent.py]: drop

**Comment by alexholdenmiller** [comment]
well, you might only want to use vocab during interactive

**Comment by stephenroller** [comment]
ya but you can still give the -ecands to interactive.py, no?

**Comment by alexholdenmiller** [comment]
Oh I didn't see the ecands arg as separate from cands... yeah thanks

**Comment by bhancock8** [comment]
Yeah, I think I would also lean toward the hard error that lets them know that they need to provide candidates for interactive mode. 'vocab' lets it output something, but it's almost never the best candidate set (e.g., even for babi, they'd rather have the six possible answers, not the 40-something words in the vocab). If we back off to 'vocab' and move along, it's not too hard to imagine someone not paying attention to the one line in the log that says what happened and just being confused why their model works but is outputting short messages.

**Comment by alexholdenmiller** [comment]
how does #1277 look?

**Pull request closed without merging**